mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java

```diff
@@ -207,8 +206,8 @@ public void testBooleanType() {
   @Test
   public void testDateType() {
     SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
-    Date gmtDate = new Date(LocalDate.of(2015, 11, 12).atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli());
-    SearchArgument arg = builder.startAnd().equals("date", PredicateLeaf.Type.DATE, gmtDate).end().build();
+    Date date = Date.valueOf("2015-11-12");
```

Comment:
We need to avoid using classes that will change behavior based on time zone to parse the value of a date or timestamp. This method was purposely not used. The problem here is probably that although the instant value is correct, the Date uses the JVM time zone somehow.

Reply:
I had a feeling this method was purposely not used, but I wasn't sure of a better option so I figured I'd open the PR and then let somebody point it out. I also figured that the flakey test was indicative of more than just a flakey test. I'll apply your suggestions this evening after work.

Reply:
I reverted this to the previous code. It's running on my local now with the patch for `daysFromDate`.